docs/messaging/kafka.md

```diff
@@ -129,40 +129,49 @@ If an intermediary broker is present, `service.name` and `peer.service` will not
-For this reason, receiving (Span Rcv1) is the parent of both processing (Span Proc1) and producing a new message (Span Prod2).
-The span representing message receiving (Span Rcv1) should not set `messaging.operation.type` to `receive`,
-as it does not only receive the message but also converts the input message to something suitable for the processing operation to consume and creates the output message from the result of processing.
-
-```
-Process P:  | Span Prod1 |
---
-Process CA:              | Span Rcv1 |
-                                | Span Proc1 |
-                                  | Span Prod2 |
---
-Process CB:                           | Span Rcv2 |
+In this example, the producer publishes a message to a topic T on Apache Kafka.
+Consumer receives the message, processes it and commits the offset.
+
+Frameworks such as Quarkus and Spring Boot provide integrations with Kafka allowing to
+configure and instrument processing callbacks, so corresponding instrumentations should create "Process"
+spans in addition to "Receive" spans created by Kafka instrumentations for polling calls.
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  P[Span Send T]
+  end
+  subgraph CONSUMER
+  direction TB
+  R1[Span Poll T]
+  R2[Span Process T]
+  R3[Span Commit T]
+  end
+
+  P-. link .-R1;
+  P-. link .-R2;
+  R2-- parent ---R3;
+
+  classDef normal fill:green
+  class P,R1,R2,R3 normal
+  linkStyle 0 color:green,stroke:green
+  linkStyle 1 color:green,stroke:green
 ```
 
-| Field or Attribute | Span Prod1 | Span Rcv1 | Span Proc1 | Span Prod2 | Span Rcv2 |
-|-|-|-|-|-|-|
-| Span name | `"T1 publish"` | `"T1 receive"` | `"T1 process"` | `"T2 publish"` | `"T2 receive`" |
-| Parent |  | Span Prod1 | Span Rcv1 | Span Rcv1 | Span Prod2 |
-| Links |  |  | |  |  |
-| SpanKind | `PRODUCER` | `CONSUMER` | `CONSUMER` | `PRODUCER` | `CONSUMER` |
-| Status | `Ok` | `Ok` | `Ok` | `Ok` | `Ok` |
-| `peer.service` | `"myKafka"` |  |  | `"myKafka"` |  |
-| `service.name` |  | `"myConsumer1"` | `"myConsumer1"` |  | `"myConsumer2"` |
-| `messaging.system` | `"kafka"` | `"kafka"` | `"kafka"` | `"kafka"` | `"kafka"` |
-| `messaging.destination.name` | `"T1"` | `"T1"` | `"T1"` | `"T2"` | `"T2"` |
-| `messaging.operation.type` |  |  | `"process"` |  | `"receive"` |
-| `messaging.client.id` |  | `"5"` | `"5"` | `"5"` | `"8"` |
-| `messaging.kafka.message.key` | `"myKey"` | `"myKey"` | `"myKey"` | `"anotherKey"` | `"anotherKey"` |
-| `messaging.kafka.consumer.group` |  | `"my-group"` | `"my-group"` |  | `"another-group"` |
-| `messaging.kafka.destination.partition` | `"1"` | `"1"` | `"1"` | `"3"` | `"3"` |
-| `messaging.kafka.message.offset` | `"12"` | `"12"` | `"12"` | `"32"` | `"32"` |
+| Field or Attribute | Span Send | Span Poll | Span Process | Span Commit |
+|-|-|-|-|-|
+| Span name | `"send T"` | `"poll T"` | `"process T"` | `"commit T"` |
+| Parent |  |  |  (optional) Span Publish  | Span Process |
```

Comment:
```suggestion
| Parent |  |  |  (optional) Span Send| Span Process |
```
Or even better

```suggestion
| Parent |  |  |  (optional) `"send T"` | Span Process |
```

No?

Reply:
Or align the names with the diagram, so it's easy to follow both

Reply:
thanks, done!